screens/CreateGame.js

```diff
@@ -0,0 +1,257 @@
+async function getAllPlayers() {
+  let userEmail = "Test@gmail.com";
+  const allPlayers = await DataStore.query(Player, (p) => p.email.ne(userEmail) );
+  return allPlayers;
+}
+
+//gets current user
+async function getPlayer(){
+    response = await Auth.currentUserInfo();
+    const player = await DataStore.query(Player, (p) => p.email.eq(response.attributes.email));
+    return player[0];
+}
+//gets the location for the game based on what the user specified 
+async function getLocation(locationName) {
+  const location = await DataStore.query(Location, (l) => l.name.eq(locationName));
+  return location;
+}
+
+//saves a name game or returns an error
+async function storeGame(gameName, gameDescription, epochDate, gameSkillLevel, organizer, location) {
+
+    //form field validation
+    // if game name is blank default name is generated
+    if([gameName].includes("")){
+        gameName = "Pickup game at " + location[0].name;
+    }
+    //if no game description is provided then set empty string
+    if([gameDescription].includes("")){
+        gameDescription = "";
+    }
+    //if no location is selected display error message
+    if([gameLocation].includes("")){
+        setShowError(true);
+        setErrorMessage("Please select a location");
+        return;
+    }
+    //if gameskilllevel is not specified set to ANY as default
+    if([gameSkillLevel].includes("")){
+        gameSkillLevel = "ANY";
+    }
+    //check if specified datetime is in the past 
+    epochNow = Math.floor(Date.now() / 1000);
+    if(epochDate <= epochNow){
+        setShowError(true);
+        setErrorMessage("Please select a valid future date and time");
+        return;
+    }
+    setShowError(false);
+    navigation.navigate("HomeScreen")
+    try {
+    const game = await DataStore.save(
+      new Game({
+        name: gameName,
+        description: gameDescription,
+        location: (await location),
+        datetime: epochDate,
+        skill_level: gameSkillLevel,
+        organizer: (await organizer).id, 
+        location_id: location.id
+      })
+    );
+    return game;
+  }
+
+  catch (error) {
+    console.log('Error saving game', error);
+    return null;
+  }
+
+}
+
+//saves game player that represents each and every game and player association
+async function storeGamePlayers(gameId, invitedPlayers) {
+
+    for (let i = 0; i < (await invitedPlayers).length; i++) {
+      try {
+        const gamePlayer = await DataStore.save(
+          new GamePlayer({
+            player_id: invitedPlayers[i].id,
+            game_id: gameId,
+            rsvp: Rsvp.PENDING,
+            invited: true
+          })
+        )
+      }
+      catch (error) {
+        console.log("error: ", error, "storing player: ", invitedPlayers[i]);
+      }
+    }
+}
+
+//function to create the game
+async function create(){
+    setLoading(true);
+    //convert chosen date to epoch timestamp in seconds
+    epochDate = Math.floor(chosenDate.getTime() / 1000);
```

Comment:
You can do input validation before the setLoading(true) call. This prevents the need to do API calls before we know if the game is valid or not.